### ISSUE TITLE
Improvements for Map page

### ISSUE THREAD

**Issue by Whaith** [hep7agon/city-feedback-hub#127]
- [x] Show (Spinner, or text in apply button) when the map is updated. Related: #118 (**Samuli**)
- [x] Disable apply button while updating. Related: #118  (**Samuli**) 
- [x] Prevent choosing start date after end date? (**Samuli**)
- [x] Pan so that the pop-up balloon is visible after highlighting (**Samuli**)
- [x] Have the default query match the selections in the UI (**Samuli**)
- [x] Disable user geolocation flood (**Samuli**)
- [x] Optimize speed (backend task)
- [x] Update heatmap to match the selected filter params. Now updates only after zoom!
- [x] Add a legend (already done in form map)
- [x] Remove leftside rounded corners in slider button

**Comment by obscurage** [comment]
What does "Optimize speed" task mean? @VitalyIvanov @Whaith ?
